In [2]:
#Importing pandas library
import pandas as pd

### Step 1: Loading the 'DPQ_J.CSV', which I was already converted

In [3]:
mydatafile1 = pd.read_csv("DPQ_J.csv")

### Step 2: Checking the dataset Shape

In [4]:
print("Shape of the dataset is:", mydatafile1.shape)

Shape of the dataset is: (5533, 11)


**Comment:** The **.shape** attribute shows how many rows and columns are present in the dataset. Here our dataset contains 5533 number of participants and 11 variables.

### Step 3: Displaying first five rows

In [5]:
mydatafile1.head()

SEQN        DPQ010        DPQ020        DPQ030        DPQ040  \
0  93705.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
1  93706.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
2  93708.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3  93709.0           NaN           NaN           NaN           NaN   
4  93711.0  1.000000e+00  5.397605e-79  1.000000e+00  5.397605e-79   

         DPQ050        DPQ060        DPQ070        DPQ080        DPQ090  \
0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
1  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
2  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3           NaN           NaN           NaN           NaN           NaN   
4  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   

         DPQ100  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4  5.397605e-79

**Comment:** The **.head()** attribute used to display the first five rows of the dataset.
It helps confirming the data loaded correctly and demonstrates how the variables are named and coded.
Columns include:

-- SEQN: Participant ID
-- DPQ010–DPQ090: PHQ-9 items (e.g., little interest, feeling down, tiredness, sleep issues, appetite, etc.)
-- DPQ100: Difficulty caused by those problems/items

### Step 4: Checking data types

In [6]:
mydatafile1.dtypes

SEQN      float64
DPQ010    float64
DPQ020    float64
DPQ030    float64
DPQ040    float64
DPQ050    float64
DPQ060    float64
DPQ070    float64
DPQ080    float64
DPQ090    float64
DPQ100    float64
dtype: object

**Comment:** All DPQ_J dataset variables, including participant number (SEQN) and PHQ-9 questions (DPQ010–DPQ100), are of type float64.
This is such that the variables are numeric with missing values (NaN), so pandas defaults to using the floating-point data type.

1. SEQN is intended to be an integer identifier, but is float64 due to the management of missing and mixed data.

2. Each DPQ variable is one unique PHQ-9 depression screener response, numerically coded 0 (not at all) to 3 (nearly every day).

3. Since all the variables share the same data type (float64), it ensures equal numeric operations, later on, which will be advantageous for statistical analysis and scoring.




### Step 5: Checking missing values

In [7]:
mydatafile1.isnull().sum()

SEQN         0
DPQ010     439
DPQ020     440
DPQ030     440
DPQ040     441
DPQ050     441
DPQ060     442
DPQ070     442
DPQ080     442
DPQ090     443
DPQ100    2171
dtype: int64

**Comment:** Missing value summary reports that there are no missing values in the participant identifier - SEQN, which is a guarantee that every record in the dataset has a unique ID.
Though all PHQ-9 items (DPQ010–DPQ090) were missing responses around 439–443, meaning that a very small number of participants omitted one or more questions.

Variable DPQ100, which measured how difficult these problems made it to do daily activities, has the highest missing responses which are 2,171 cases.
This is to be expected as the difficulty question is asked of those participants who have answered that they have experienced at least one symptom, i.e., those who responded "0" on all items were not asked this follow-up question.

In general, the missingness pattern suggests partial nonresponse instead of random data loss.
These gaps must be handled with caution which is for instance, by missing record exclusion or imputation before determining total PHQ-9 scores.

### Step 6: Checking variable names

In [8]:
columns = mydatafile1.columns.tolist()
columns

['SEQN',
 'DPQ010',
 'DPQ020',
 'DPQ030',
 'DPQ040',
 'DPQ050',
 'DPQ060',
 'DPQ070',
 'DPQ080',
 'DPQ090',
 'DPQ100']

**Comment:** Making sure the dataset includes the expected PHQ-9 items 'DPQ010–DPQ090' and the difficulty item 'DPQ100', additionally the participant ID 'SEQN'.

### Asserting the expected PHQ-9

In [12]:
expected_columns = ["SEQN","DPQ010","DPQ020","DPQ030","DPQ040","DPQ050",
            "DPQ060", "DPQ070","DPQ080","DPQ090","DPQ100"]

missing_in_datafile = [c for c in expected_columns if c not in mydatafile1.columns]
extra_in_datafile   = [c for c in mydatafile1.columns if c not in expected_columns]

print("Missing expected columns:", missing_in_datafile)
print("Extra columns present:", extra_in_datafile)


Missing expected columns: []
Extra columns present: []


**Comment:** Checked if there were any extra or missing columns in the datafile and yes, everything looks great. This is just a suggestable step before merging with another.

### PHQ-9 Variable Map (DPQ_J)

| Variable | PHQ-9 Item (last 2 weeks) |
|---------|----------------------------|
| `DPQ010` | Little interest or pleasure in doing things |
| `DPQ020` | Feeling down, depressed, or hopeless |
| `DPQ030` | Trouble falling/staying asleep or sleeping too much |
| `DPQ040` | Feeling tired or having little energy |
| `DPQ050` | Poor appetite or overeating |
| `DPQ060` | Feeling bad about yourself / failure or letting yourself or your family down |
| `DPQ070` | Trouble concentrating on things |
| `DPQ080` | Moving/speaking slowly or being fidgety/restless |
| `DPQ090` | Thoughts that you would be better off dead, or of hurting yourself |
| `DPQ100` | If any problems above: how difficult have these made things at work, home, or with other people? |


### Observations(DPQ_J)

- The database has **5533 participants** and **11 variables**.
- All PHQ-9 items (`DPQ010`–`DPQ090`) and the question about functional difficulty (`DPQ100`) are present.
- All columns are `float64`, as one might expect from **missing values (NaN)** in self-report items; `SEQN` may also be `float64`.
- Missingness is rather low between PHQ-9 items; `DPQ100` will tend to have more missing because it is **conditionally asked**.
- It is compatible with the NHANES DPQ codebook and can be appended to `DEMO_J` on **SEQN** in the next notebook.